libfreerdp/core/settings.c

```diff
@@ -277,6 +277,199 @@ void freerdp_settings_print_warnings(const rdpSettings* settings)
+			                         .bottom = monitor->y + monitor->height };
+
+		if (rectangles_intersects(&rect1, &rect2))
+		{
+			WLog_ERR(
+			    TAG,
+			    "Mulitimonitor mode requested, but local layout has gaps or overlapping areas!");
+			WLog_ERR(TAG, "Please reconfigure your local monitor setup so that thre are no gaps or "
+			              "overlapping areas!");
+			return TRUE;
+		}
+	}
+	return FALSE;
+}
+
+static BOOL monitor_has_gaps(const rdpSettings* settings, UINT32 start, UINT32 count,
+                             const rdpMonitor* compare)
+{
+	BOOL hasNeighbor = FALSE;
+	for (UINT32 x = 0; x < count; x++)
+	{
+		if (x == start)
+			continue;
+
+		const rdpMonitor* monitor =
+		    freerdp_settings_get_pointer_array(settings, FreeRDP_MonitorDefArray, x);
+
+		if (compare->x == monitor->x + monitor->width)
+			hasNeighbor = TRUE;
+		if (compare->y == monitor->y + monitor->height)
+			hasNeighbor = TRUE;
+		if (compare->x + compare->width == monitor->x)
+			hasNeighbor = TRUE;
+		if (compare->y + compare->height == monitor->y)
+			hasNeighbor = TRUE;
+	}
+	if (!hasNeighbor)
+	{
+		WLog_ERR(TAG,
+		         "Monitor configuration has gaps! Monitor %" PRIu32 " does not have any neighbor",
+		         start);
+	}
+	return !hasNeighbor;
+}
+
+static BOOL freerdp_settings_client_monitors_have_gaps(const rdpSettings* settings)
+{
+	const UINT32 count = freerdp_settings_get_uint32(settings, FreeRDP_MonitorCount);
+	if (count <= 1)
+		return FALSE;
+
+	for (UINT32 x = 0; x < count; x++)
+	{
+		const rdpMonitor* monitor =
+		    freerdp_settings_get_pointer_array(settings, FreeRDP_MonitorDefArray, x);
+		if (monitor_has_gaps(settings, x, count, monitor))
+			return TRUE;
+	}
+	return FALSE;
+}
+
+static BOOL freerdp_settings_client_monitors_overlap(const rdpSettings* settings)
+{
+	const UINT32 count = freerdp_settings_get_uint32(settings, FreeRDP_MonitorCount);
+	for (UINT32 x = 0; x < count; x++)
+	{
+		const rdpMonitor* monitor =
+		    freerdp_settings_get_pointer_array(settings, FreeRDP_MonitorDefArray, x);
+		if (monitor_operlaps(settings, x + 1, count, monitor))
+			return TRUE;
+	}
+	return FALSE;
+}
+
+/* See [MS-RDPBCGR] 2.2.1.3.6.1 for details on limits
+ * https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/c3964b39-3d54-4ae1-a84a-ceaed311e0f6
+ */
+static BOOL freerdp_settings_client_monitors_check_primary_and_origin(const rdpSettings* settings)
+{
+	const UINT32 count = freerdp_settings_get_uint32(settings, FreeRDP_MonitorCount);
+	BOOL havePrimary = FALSE;
+	BOOL foundOrigin = FALSE;
+	BOOL rc = TRUE;
+
+	struct
+	{
+		INT32 x;
+		INT32 y;
+		INT32 width;
+		INT32 height;
+	} bounds = { 0 };
+
+	for (UINT32 x = 0; x < count; x++)
+	{
+		const rdpMonitor* monitor =
+		    freerdp_settings_get_pointer_array(settings, FreeRDP_MonitorDefArray, x);
+		if (monitor->x < bounds.x)
+		{
+			bounds.x = monitor->x;
+			bounds.width += monitor->width;
+		}
+		if (monitor->x >= bounds.x + bounds.width)
```

Comment:
I don't think this condition (and the y equivalent) is correct i.e.:

![image](https://github.com/user-attachments/assets/d9496091-6eb3-4fb6-b4b3-ff190eeb8c93)

The second monitor's Y (40) won't be larger than the first monitor's Y + height (i.e. 0 + 800) so the bounding rectangle keeps the size of just the first monitor.